_gsocproposals/2020/proposal_MCnetSherpa.md

```diff
@@ -0,0 +1,50 @@
+
+In the ongoing quest for increased precision of our understanding and modelling of the physics involved in the proton collisions, ever higher orders of the perturbation expansion must be evaluated.  This results in ever more complicated expressions for the transition amplitudes and an ever increasing number of final state particles and therefore the dimensionality of the phase space integration.  It is therefore not surprising that this first step of the event simulation is the most intensive source of CPU consumption and becoming the bottleneck for the full exploitation of future LHC results.
+
+This project will be co-supervised by Frank Krauss, SHERPA's primary author, and Tobias Weinzierl, an expert in HPC computations and their performance optimization.  The project will focus on first steps to increase the efficiency of the SHERPA event generator, and in particular of its description of the hard process.  In a first step, the successful students will analyse the code and derive a detailed performance profile. Given the embarrassingly parallel nature of the underlying algorithmics, the studies will focus on single-node performance. In Durham, Intel's Amplifier VTune and the Allinea tools are available for this task, as well as a suite of open source helpers such as Likwid.
+
+With a detailed performance profile at hand, the students will work out an exascale optimisation roadmap. We expect this roadmap to discuss mainly generic strategies mapped onto the particular application domain. This includes the flattening and serialisation of data structures to avoid scattered memory accesses, the reordering of data accesses, where the multi-shoot character of Monte Carlo algorithms usually provides ample space for relatively simple performance optimization.
+
+## Task ideas
+
+The following tasks are envisaged:
+
+ * Overall performance profile of the SHERPA code
+ * Detailed performance profile of the hard process simulation in the SHERPA code, with emphasis on data structures and access
+ * First draft of an exascale roadmap to efficiently deploy SHERPA on future HPC architectures
+
+## Expected results
+
+ * Improvements to SHERPA's performance, especially for the integration of cross sections for processes with multi-particle final-states
+ * First draft of an Exascale Roadmap for SHERPA
+
+## Requirements
+
+ * C++
+ * Profiling
+ * Performance optimisation
+ * Multithreading

```

Comment:
You will find not many candidates with experience on performance optimisation, and it is hard to demonstrate as skill.

Reply:
True , but that is what's needed and it's a code task. If the candidates aren't available, that's life